source/particle/world.cc

```diff
@@ -158,19 +158,36 @@ namespace aspect
     {
       // If we do not write output
       // return early with the number of particles that were advected
-      if (!output)
+      if (output.empty())
         return "";
 
       TimerOutput::Scope timer_section(this->get_computing_timer(), "Particles: Output");
       const double output_time = (this->convert_output_to_years() ?
                                   this->get_time() / year_in_seconds :
                                   this->get_time());
 
-      const std::string filename = output->output_particle_data(particles,
-                                                                property_manager->get_data_info(),
-                                                                output_time);
+      std::vector<std::string> output_suffixes;
 
-      return filename;
+      const std::string output_directory = output[0].get()->get_particle_output_location();
+      const std::string file_index = output[0].get()->get_file_index();
```

Comment:
Here you assume that `file_index` is the same for all output plugins. However, inside of the serialization of the plugins you seem to allow that they can be enabled or disabled during restarts (leading to different file indices in the different plugins). Would this lead to problems? At least it would cause the screen output and statistics output of the file names to be not consistent with the actual filenames of some plugins. Is there an easy fix for that? If not, how can we best document this behavior?